source/docs/software/wpilib-tools/path-planning/wpilib-trajectories/ramsete.rst

```diff
@@ -0,0 +1,55 @@
+
+.. note:: Gains of ``2.0`` and ``0.7`` for ``b`` and ``zeta`` have been tested repeatedly to produce desirable results.
+
+Getting adjusted velocities
+---------------------------
+The Ramsete controller returns "adjusted velocities" so that the when the robot tracks these velocities, it accurately reaches the goal point. The controller should be updated periodically with the new goal. The goal comprises of a desired pose, desired linear velocity, and desired angular velocity. Furthermore, the current position of the robot should also be updated periodically. The controller uses these four arguments to return the adjusted linear and angular velocity. Users should command their robot to these linear and angular velocities to achieve optimal trajectory tracking.
+
+.. note:: The "goal pose" represents the position that the robot should be at at a particular timestep when tracking the trajectory. It does NOT represent the final endpoint of the trajectory.
+
+The controller can be updated using the ``Calculate`` (C++) / ``calculate`` (Java) method. There are two overloads for this method. Both of these overloads accept the current robot position as the first parameter. For the other parameters, one of these overloads takes in the goal as three separate parameters (pose, linear velocity, and angular velocity) whereas the other overload accepts a ``Trajectory.State`` object, which contains information about the goal pose. For its ease, users should use the latter method when tracking trajectories.
+
+.. tabs::
+
+   .. code-tab:: java
+
+      var goal = trajectory.sample(3.4); // sample the trajectory at 3.4 seconds from the beginning
+      ChassisSpeeds adjustedSpeeds = controller.calculate(currentRobotPose, goal);
+
+
+   .. code-tab:: c++
+
+      const auto goal = trajectory.Sample(3.4_s); // sample the trajectory at 3.4 seconds from the beginning
+      ChassisSpeeds adjustedSpeeds = controller.Calculate(currentRobotPose, goal);
+
+Using the adjusted velocities
+-----------------------------
+The adjusted velocities are of type ``ChassisSpeeds``, which contains a ``vx`` (linear velocity in the forward direction), a ``vy`` (linear velocity in the sideways direction), and an ``omega`` (angular velocity around the center of the robot frame). Because the Ramsete controller is a controller for non-holonomic robots (robots which cannot move sideways), the adjusted speeds object has a ``vy`` of zero.
+
+The returned adjusted speeds can be converted to usable speeds using the kinematics classes for your drivetrain type. For example, the adjusted velocities can be converted to left and right velocities for a differential drive using a ``DifferentialDriveKinematics`` object.
+
+.. tabs::
+
+   .. code-tab:: java
+
+      ChassisSpeeds adjustedSpeeds = controller.calculate(currentRobotPose, goal);
```

Comment:
Should you be recreating this object on every robot loop? Again, what object is "wheelSpeeds"

Reply:
> Should you be recreating this object on every robot loop?
Addressed this in previous comment.
> Again, what object is "wheelSpeeds"
Pushed a new commit that specifies the type explicitly.